src/sql.rs

```diff
@@ -306,37 +306,49 @@ impl Sql {
         }
     }
 
-    /// Locks the write transactions mutex.
-    /// We do not make all transactions
-    /// [IMMEDIATE](https://www.sqlite.org/lang_transaction.html#deferred_immediate_and_exclusive_transactions)
-    /// for more parallelism -- at least read transactions can be made DEFERRED to run in parallel
-    /// w/o any drawbacks. But if we make write transactions DEFERRED also w/o any external locking,
-    /// then they are upgraded from read to write ones on the first write statement. This has some
-    /// drawbacks:
-    /// - If there are other write transactions, we block the thread and the db connection until
-    ///   upgraded. Also if some reader comes then, it has to get next, less used connection with a
-    ///   worse per-connection page cache.
-    /// - If a transaction is blocked for more than busy_timeout, it fails with SQLITE_BUSY.
-    /// - Configuring busy_timeout is not the best way to manage transaction timeouts, we would
```

Comment:
This one is removed, but also looks important, no?

Reply:
This was:
```
    /// - Configuring busy_timeout is not the best way to manage transaction timeouts, we would
    ///   prefer it to be integrated with Rust/tokio asyncs. Moreover, SQLite implements waiting
    ///   using sleeps.
```
which I replaced with:
```
    /// - SQLite's locking mechanism is non-async, blocking a thread
    /// - SQLite's locking mechanism just sleeps in a loop, which is really inefficient
```
and:
```
    /// 1. As mentioned above, SQLite's locking mechanism is non-async and sleeps in a loop.
```
Is there some important information missing now which you'd like to have included?

Reply:
Hm, SQLite's busy_timeout is configured per-connnection, so looks like we can control a timeout with enough precision. But it's a question how to cancel waiting for this timeout. Even if it's possible, better to use what tokio offers for that. If u want, u can add this also to the comment

Reply:
I don't think I fully understand, which probably also means that it's hard to put into a comment that should be understandable by people who didn't follow the background story. Since it didn't seem that important to you, I'm just going to enable auto-merge on this PR.